cpp/src/parquet/schema_test.cc

```diff
@@ -1691,6 +1691,18 @@ TEST(TestSchemaNodeCreation, FactoryExceptions) {
   // Primitive too small for given precision ...
   ASSERT_ANY_THROW(PrimitiveNode::Make("decimal", Repetition::REQUIRED,
                                        DecimalLogicalType::Make(16, 6), Type::INT32));
+  ASSERT_ANY_THROW(PrimitiveNode::Make("decimal", Repetition::REQUIRED,
+                                       DecimalLogicalType::Make(10, 9), Type::INT32));
+  ASSERT_ANY_THROW(PrimitiveNode::Make("decimal", Repetition::REQUIRED,
+                                       DecimalLogicalType::Make(17, 18), Type::INT64));
```

Comment:
this looks like it should fail for other reasons (scale greater than precision).

Reply:
Yes, added a comment and a new test with precision=19.